src/syft_rds/client/rds_client.py

```diff
@@ -143,80 +170,128 @@ def datasets(self) -> list[Dataset]:
         """
         return self.dataset.get_all()
 
-    def get_default_config_for_job(self, job: Job) -> JobConfig:
+    def _get_config_for_job(self, job: Job) -> JobConfig:
         user_code = self.user_code.get(job.user_code_id)
         dataset = self.dataset.get(name=job.dataset_name)
-        runtime = dataset.runtime or self.config.runner_config.runtime
+        runtime = self.runtime.get(job.runtime_id)
         runner_config = self.config.runner_config
-        return JobConfig(
-            function_folder=user_code.local_dir,
-            args=[user_code.entrypoint],
+        job_config = JobConfig(
             data_path=dataset.get_private_path(),
+            function_folder=user_code.local_dir,
             runtime=runtime,
+            args=[user_code.entrypoint],
             job_folder=runner_config.job_output_folder / job.uid.hex,
             timeout=runner_config.timeout,
-            use_docker=runner_config.use_docker,
         )
+        return job_config
+
+    def update_job_status(self, job_update: JobUpdate, job: Job) -> Job:
+        new_job = self.rpc.job.update(job_update)
+        return job.apply_update(new_job)
 
     def run_private(
         self,
         job: Job,
-        config: Optional[JobConfig] = None,
         display_type: str = "text",
         show_stdout: bool = True,
         show_stderr: bool = True,
     ) -> Job:
         if job.status == JobStatus.rejected:
             raise ValueError(
-                "Cannot run rejected job, "
-                "if you want to override this, "
-                "set job.status to something else"
+                "Cannot run rejected job. "
+                "If you want to override this, set `job.status` to something else."
             )
-
-        config = config or self.get_default_config_for_job(job)
-
-        logger.warning("Running job without docker is not secure")
-        return_code = self._run(
-            config=config,
-            display_type=display_type,
-            show_stdout=show_stdout,
-            show_stderr=show_stderr,
+        logger.debug(f"Running job '{job.name}' on private data")
+        job_config: JobConfig = self._get_config_for_job(job)
+        result = self._run(
+            job_config,
+            job,
+            display_type,
+            show_stdout,
+            show_stderr,
         )
-        job_update = job.get_update_for_return_code(return_code)
-        new_job = self.rpc.job.update(job_update)
-        return job.apply_update(new_job)
+        return self._handle_result(result, job)
```

Comment:
How does handle_result work for a non-blocking job? what is the result in that case?